jp/match.go

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2024, Peter Ohler, All rights reserved.
+
+package jp
+
+// PathMatch returns true if the provided path would match the target
+// expression. The path argument is expected to be a normalized path with only
+// elements of Root ($), At (@), Child (string), or Nth (int). A Filter
+// fragment in the target expression will match any value in path since it
+// requires data from a JSON document to be evaluated. Slice fragments always
+// return true as long as the path element is an Nth.
+func PathMatch(target, path Expr) bool {
+	if 0 < len(target) {
+		switch target[0].(type) {
+		case Root, At:
+			target = target[1:]
+		}
+	}
+	if 0 < len(path) {
+		switch path[0].(type) {
+		case Root, At:
+			path = path[1:]
+		}
+	}
+	for i, f := range target {
+		if len(path) == 0 {
+			return false
+		}
```

Comment:
Why waiting to be in the for loop to do this while it can be done before?

Reply:
The path gets modified during the loop over the target. When the path is empty then nothing has matched to that point.

Reply:
I missed it, indeed